rclcpp/include/rclcpp/parameter.hpp

```diff
@@ -101,61 +109,78 @@ class ParameterVariant
     return value_;
   }
 
+  template<ParameterType type>
+  typename std::enable_if<type == ParameterType::PARAMETER_INTEGER, int64_t>::type
+  get_value() const
+  {
+    if (value_.parameter_type != rcl_interfaces::ParameterType::PARAMETER_INTEGER) {
+      // TODO: use custom exception
+      throw std::runtime_error("Invalid type");
+    }
+    return value_.integer_value;
+  }
+  template<ParameterType type>
+  typename std::enable_if<type == ParameterType::PARAMETER_DOUBLE, double>::type
+  get_value() const
+  {
+    if (value_.parameter_type != rcl_interfaces::ParameterType::PARAMETER_DOUBLE) {
+      // TODO: use custom exception
+      throw std::runtime_error("Invalid type");
+    }
+    return value_.double_value;
+  }
+  template<ParameterType type>
+  typename std::enable_if<type == ParameterType::PARAMETER_STRING, const std::string &>::type
+  get_value() const
+  {
+    if (value_.parameter_type != rcl_interfaces::ParameterType::PARAMETER_STRING) {
+      // TODO: use custom exception
+      throw std::runtime_error("Invalid type");
+    }
+    return value_.string_value;
+  }
+  template<ParameterType type>
+  typename std::enable_if<type == ParameterType::PARAMETER_BOOL, bool>::type
+  get_value() const
+  {
+    if (value_.parameter_type != rcl_interfaces::ParameterType::PARAMETER_BOOL) {
+      // TODO: use custom exception
+      throw std::runtime_error("Invalid type");
+    }
+    return value_.bool_value;
+  }
+  template<ParameterType type>
+  typename std::enable_if<type == ParameterType::PARAMETER_BYTES, std::vector<uint8_t>>::type
+  get_value() const
+  {
+    if (value_.parameter_type != rcl_interfaces::ParameterType::PARAMETER_BYTES) {
+      // TODO: use custom exception
+      throw std::runtime_error("Invalid type");
+    }
+    return value_.bytes_value;
+  }
+
+  int64_t as_int() const {return get_value<ParameterType::PARAMETER_INTEGER>(); }
+
+  double as_double() const {return get_value<ParameterType::PARAMETER_DOUBLE>(); }
+
+  const std::string & as_string() const {return get_value<ParameterType::PARAMETER_STRING>(); }
```

Comment:
With `get_value<>` not returning a const reference but a temporary object this won't work I guess?